### ISSUE TITLE
Icons (and Their Matching Commands ) of view/title Do not Update Upon Changes in Workspace settings.json

### ISSUE THREAD

**Issue by TomHermanSAP** [eclipse-theia/theia#9075]
### Bug Description:

The Maven for Java extension uses 2 icons in its view/title menu:
https://github.com/microsoft/vscode-maven/blob/master/package.json#L290-L298

<img src="https://user-images.githubusercontent.com/54273024/108079283-55553a00-7077-11eb-82bc-2190951b0019.png" alt="flat" height=100/>

<img src="https://user-images.githubusercontent.com/54273024/108079411-7584f900-7077-11eb-8eef-be84ab6f9c5f.png" alt="hierarchical" height=100/>

The "flat" and "hierarchical" icons should replace one another upon clicking on them:

When the user clicks on the icon, a change is made to the Workspace Configuration:
https://github.com/microsoft/vscode-maven/blob/master/src/Settings.ts#L20-L26

The icon at view/title does not change after clicking (and so is the matching command it exposes).

When removing the matching configuration "maven.view" from the Workspace settings.json, and pasting it in the User settings.json, upon manual change of the value (between "flat" and "hierarchical"), the icon changes accordingly and as expected:

<img src="https://user-images.githubusercontent.com/54273024/108079956-0bb91f00-7078-11eb-845d-401df253769b.png" alt="flat" height=40/>

<img src="https://user-images.githubusercontent.com/54273024/108079894-fcd26c80-7077-11eb-97a2-af53aa07eaa2.png" alt="hierarchical" height=40/>

Bottom line - It looks like the "when" clause in the package.json that uses "config":

`"when": "view == mavenProjects && config.maven.view != flat",`
(from https://github.com/microsoft/vscode-maven/blob/master/package.json#L291)

Listens to changes of the User configurations and not the Workspace configurations.

### Steps to Reproduce:

1. Open https://gitpod.io/#https://github.com/gitpod-io/spring-petclinic
2. Install the "Maven for Java" Extension.
3. In the Maven menu (in Explorer Tab), click on the icon "Switch to hierarchical view" - NOTHING HAPPENS.

4. Open File -> Settings -> Preferences.
5. In Preferences tab -> Workspaces -> Open settings.json file.
6. Remove `"maven.view": "hierarchical"`
7. In Preferences tab -> User-> Open settings.json file.
8. Add `"maven.view": "hierarchical"`. See Icon. Change value from "hierarchical" to "flat". See how the icon changes.

- Operating System: gitpod
- Theia Version: gitpod

**Comment by amiramw** [comment]
@TomHermanSAP @danarad05 do you think it is related to https://github.com/eclipse-theia/theia/issues/8757?

**Comment by TomHermanSAP** [comment]
@amiramw @danarad05 I don't think it is related to https://github.com/eclipse-theia/theia/issues/8757.

If I keep the pointer on the title bar and change the value in the settings.json:
1. The Tree items' order changes accordingly and as wanted.
2. The icon does not change.

Once I move the pointer away from the title bar and return it back again to the title bar - the icon changes as wanted.

**Comment by vince-fugnitto** [comment]
@TomHermanSAP I believe it is a duplicate of https://github.com/eclipse-theia/theia/issues/6082, if so can you please close the old issue?

**Comment by TomHermanSAP** [comment]
Hi @vince-fugnitto,
If the issue creator, @svenefftinge, meant that they can see both icons side by side, this is not the case anymore (as seen in the screenshots in this issue). 

Therefore the old issue (https://github.com/eclipse-theia/theia/issues/6082) can be closed (I do not have permissions to close the old issue).

**Comment by danarad05** [comment]
> Once I move the pointer away from the title bar and return it back again to the title bar - the icon changes as wanted.

That part is solved by #8757.

> "when": "view == mavenProjects && config.maven.view != flat",
Listens to changes of the User configurations and not the Workspace configurations

I'm guessing you are right - that config is referencing User settings.

**Comment by balajiv113** [comment]
@danarad05 - I looked into this issue, the getValue of configuration service was the issue. 
resourceUri is set as false in the below mentioned line.
https://github.com/eclipse-theia/theia/blob/f3a15db2e364549922a843f195371f5a19a1a403/packages/monaco/src/browser/monaco-frontend-module.ts#L158
Due to this preference proxy is returning the default value which is flat.

I will raise a PR for fixing this part. After this fix the toggle will work fine (icons will be updated only after hover out and hover in on maven tab).
